modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -641,7 +641,24 @@ def to_pandas(cls, partitions):
         pandas.DataFrame
             A pandas DataFrame
         """
-        retrieved_objects = [[obj.to_pandas() for obj in part] for part in partitions]
+        retrieved_objects = cls.get_objects_from_partitions(partitions.flatten())
+        if all(
+            isinstance(obj, (pandas.DataFrame, pandas.Series))
```

Comment:
I wonder when we can receive a `pandas.Series` stored within a partition...

Reply:
I don't know for sure, but we have an explicit check for this type in the `to_pandas` function https://github.com/modin-project/modin/blob/6d5545f4a132f0efce02db66a6f5d515d4000812/modin/core/dataframe/pandas/partitioning/partition.py#L146

I just want to be aligned with this.